modules/assim.sequential/inst/WillowCreek/Download_US_Wcr.R

```diff
@@ -0,0 +1,70 @@
+#' download_US_WCr
+#'
+#' @param start_date start date for downloading the flux data
+#' @param end_date  start date for downloading the flux data
+#' @description NEE is converted to  kg C m-2 s-1. The main difference between this function and the one in the 
+#' PEcAn.data.atmosphere package is that, this function returns all the measured variables as well as date.
+#'
+#' @return a dataframe with flux data downloaded for the Willow Creek site
+#' @export
+#'
+#' @examples
+download_US_WCr <- function(start_date, end_date) {
+  base_url <- "http://co2.aos.wisc.edu/data/cheas/wcreek/flux/prelim/wcreek"
+  
+  start_year <- lubridate::year(start_date)
+  end_year <- lubridate::year(end_date)
+  
+  
+  # Reading in the data
+  suppressWarnings(
+  raw.data <- start_year:end_year %>%
+    purrr::map_df(function(syear) {
+      influx <-
+        tryCatch(
+          read.table(
+            paste0(base_url, syear, "_flux.txt"),
+            sep = "",
+            header = FALSE
+          ) %>%
+            apply(2, trimws) %>%
+            apply(2, as.character) %>%
+            data.frame(stringsAsFactors = F),
+          error = function(e) {
+            NULL
+          },
+          warning = function(e) {
+            NULL
+          }
+        )
+    }) %>%
+    mutate_all(funs(as.numeric))
+  )
+  # cleaning the -999 and replacing it with NA
+  raw.data[ which(raw.data==-999, TRUE) ] <- NA 
+  
+  #Constructing the date based on the columns we have
+  raw.data$date <-as.POSIXct(paste0(raw.data$V1,"/",raw.data$V2,"/",raw.data$V3," ", raw.data$V4 %>% as.integer(), ":",(raw.data$V4-as.integer(raw.data$V4))*60),
+                              format="%Y/%m/%d %H:%M", tz="UTC")
+  # Some cleaning and filtering 
+  raw.data <- raw.data %>% 
+      select(-V1,-V2,-V3,-V4,-V5) %>%
+    filter(date <=end_date, date >=start_date) %>%
+    `colnames<-`(c("Fjday", "SC", "FC", "NEE", "LE", "H", "U", "Flag", "Date"))
+  
+  #Unit conversion
+  raw.data$NEE <- PEcAn.utils::misc.convert(raw.data$NEE, "umol C m-2 s-1", "kg C m-2 s-1")
```

Comment:
so does this get passed somewhere else to create obs for SDA?

Reply:
Yes, we are working on a project that these tools help us to have a full automated workflow for downloading / preparing the obs data and running an SDA for Willow Creek site every day.